src/app/dataset-view/additional-components/metadata-component/components/form-components/json-kind-field/json-kind-field.component.ts

```diff
@@ -0,0 +1,30 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit } from "@angular/core";
+import { BaseField } from "../base-field";
+import { FormControl } from "@angular/forms";
+import { ReadKind } from "../../add-polling-source/add-polling-source-form.types";
+import { RadioControlType } from "../../add-polling-source/form-control.source";
+import { SetPollingSourceToolipsTexts } from "src/app/common/tooltips/tooltips.text";
+
+@Component({
+    selector: "app-json-kind-field",
+    templateUrl: "./json-kind-field.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class JsonKindFieldComponent extends BaseField implements OnInit {
+    @Input() public data: RadioControlType[];
+    public currentJsonKind: ReadKind;
+    public readKind: typeof ReadKind = ReadKind;
```

Comment:
Always use type name in original case - this is visually super confusing, we can't distinguish what is a variable and what is a type just by looking on the name:

`public ReadKind: typeof ReadKind = ReadKind`

Reply:
Fixed on `jsonReadFormat`